daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -513,6 +520,39 @@ var newTicker = func(freq time.Duration) *time.Ticker {
 	return time.NewTicker(freq)
 }
 
+var logMessagesDiscardedWarning = func(logGroupName, logStreamName string, maxBufferedEvents int, discarded int32) {
+	logrus.WithFields(logrus.Fields{
+		"driver":             name,
+		"logGroupName":       logGroupName,
+		"logStreamName":      logStreamName,
+		"messagesDiscarded":  discarded,
+		maxBufferedEventsKey: maxBufferedEvents,
+	}).Warn("Log messages are being discarded")
+}
+
+// reportDiscardedMessages periodically checks discardedMessages counter, logging a warning if it's greater than zero.
+// The warning can be useful for the user to realize that container logs are being discarded. Container logs will be
+// discarded when this driver is configured in non-blocking mode and logs arrive faster than they can be processed
+// (i.e. sent to CloudWatch).
+func (l *logStream) reportDiscardedMessages() {
+	ticker := newTicker(time.Minute)
+	defer ticker.Stop()
+	for {
+		// If logger is closed we are done
```

Comment:
So the logger could be closed but we could've missed a warning message to log that we did discard some logs, right? Do we care about that in this case?

Do we want to have a unit test in that scenario?

Reply:
Good call. I think I can just check again if there are discarded messages before returning and we should be covered since no new messages are accepted after the driver is closed.